GCRCatalogs/photoz.py

```diff
@@ -0,0 +1,128 @@
+"""
+Photo-z catalog reader
+
+This reader was designed by Yao-Yuan Mao,
+based a photo-z catalog provided by Sam Schmidt, in Feb 2019.
+"""
+
+import re
+import os
+import warnings
+import shutil
+
+import yaml
+import numpy as np
+import pandas as pd
+
+from GCR import BaseGenericCatalog
+
+__all__ = ['PhotoZCatalog']
+
+FILE_PATTERN = r'run\d\.\d+[a-z]+_PZ_tract_\d+\.h5$'
+METADATA_FILENAME = 'metadata.yaml'
+PDF_BIN_INFO = {
+    'start': 0.005,
+    'stop': 1.01,
+    'step': 0.01,
+    'decimals_to_round': 3,
+}
+
+class PhotoZCatalog(BaseGenericCatalog):
+
+    def _subclass_init(self, **kwargs):
+        self.base_dir = kwargs['base_dir']
+        self._filename_re = re.compile(kwargs.get('filename_pattern', FILE_PATTERN))
+        _metadata_filename = kwargs.get('metadata_filename', METADATA_FILENAME)
+        self._metadata_path = os.path.join(self.base_dir, _metadata_filename)
+
+        self._pdf_bin_info = kwargs.get('pdf_bin_info', PDF_BIN_INFO)
+        self.pdf_bin_centers = np.round(np.arange(
+            self._pdf_bin_info['start'],
+            self._pdf_bin_info['stop'],
+            self._pdf_bin_info['step'],
+        ), self._pdf_bin_info['decimals_to_round'])
+        self._n_pdf_bins = len(self.pdf_bin_centers)
+
+        if self._metadata_path and os.path.isfile(self._metadata_path):
+            with open(self._metadata_path, 'r') as meta_stream:
+                self._metadata = yaml.load(meta_stream)
+        else:
+            self._metadata = self.generate_metadata()
+
+        self._quantity_modifiers = {
+            'id': 'ID',
+            'pz_z_peak': 'z_peak',
+            'pz_pdf_full': '_full_pdf',
+        }
```

Comment:
OK, so should we use `photoZ` and `photoZ_peak` (or `photoZ_pest_peak`)?

Reply:
or `photoZ_mode`?

Reply:
Also, should we then just *not* provide individual columns for PDF (i.e., `pz_z0_005` etc)？

Reply:
I also noticed that in DPDD, `photoZ` 
(1) contains the bin centers, and
(2) contains log L rather than L. 

So, do we want to follow these formats too?

Reply:
I don't know.  I'm torn.

Reply:
The actual format of that 2x95 array is a little silly and deeply inefficient (not that the DPDD says things will be _stored_ that way, it's just what will be presented to the user, but still).  Repeating each z array for each object is not what we want here.

Reply:
In the interest of time, I'm going to use
- `photoZ_mode` for z_peak 
- `photoZ_pdf` for the full pdf
- `photoZ_zX_XX` for the individual columns. 

I keep the `photoZ_` prefix to be consistent with DPDD, but don't use exactly the same name since the definitions are different. 

Is that ok?

Reply:
And understanding how these should be presented and users want to use them is part of what DESC can provide usefull feedback about.

Reply:
I suggest we try:
`photoz_mode`
`photoz_pdf`

Reply:
But how should we clearly indicate what the z array is for the `photoz_pdf`?

Reply:
You want to use lower-case 'z'? 

there is `pz_pdf_bin_centers` attribute (which probably should be renamed to `photoz_pdf_bin_centers`).